src/lib/hooks/useDomainMetadata.ts

```diff
@@ -0,0 +1,60 @@
+//- React Imports
+import { useQuery } from 'react-query';
+
+//- Types Imports
+import { Metadata } from '../types/metadata';
+
+//- Util Imports
+import { parseDomainMetadata } from '../util/metadata/metadata';
+
+//- Library Imports
+import useZnsSdk from './useZnsSdk';
+
+export interface UseDomainMetadataReturn {
+	domainMetadata?: Metadata | undefined;
```

Comment:
Is it the same `domainMetadata?: Metadata;`?

Reply:
> Is it the same `domainMetadata?: Metadata;`?

Yep same thing, do either `domainMetadata?: Metadata` or `domainmetadata: Metadata | undefined`. Maybe since this is a subjective thing we could put it up to a vote and put it in our code styleguide?